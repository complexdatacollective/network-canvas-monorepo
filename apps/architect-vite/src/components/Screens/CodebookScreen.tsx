import { Button } from "@codaco/legacy-ui/components";
import Codebook from "~/components/Codebook/Codebook";
import { Layout } from "~/components/EditorLayout";
import Screen from "~/components/Screen/Screen";
import ControlBar from "../ControlBar";
import CollapsableHeader from "../Screen/CollapsableHeader";

type CodebookScreenProps = {
	layoutId?: string | null;
	onComplete?: () => void;
};

/**
 * This component acts as an index for types. i.e. Nodes and Edges,
 * and links to the EditType.
 */
const CodebookScreen = ({ layoutId = null, onComplete = () => {} }: CodebookScreenProps) => {

	const buttons = [
		<Button key="done" onClick={onComplete} color="platinum">
			Close
		</Button>,
	];

	return (
		<Screen layoutId={layoutId} footer={<ControlBar buttons={buttons} />} onComplete={onComplete}>
			<CollapsableHeader
				collapsedState={
					<div className="stage-heading stage-heading--collapsed stage-heading--shadow">
						<Layout>
							<h2>Codebook</h2>
						</Layout>
					</div>
				}
			>
				<div className="stage-heading stage-heading--inline">
					<Layout>
						<h1 className="screen-heading">Codebook</h1>
						<p>
							Below you can find an overview of the node and edge types that you have defined while creating your
							interview. Entities that are unused may be deleted.
						</p>
					</Layout>
				</div>
			</CollapsableHeader>
			<Layout>
				<Codebook />
			</Layout>
		</Screen>
	);
};


export default CodebookScreen;
