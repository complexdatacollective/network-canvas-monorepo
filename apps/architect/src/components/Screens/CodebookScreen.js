import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@codaco/ui/lib/components';
import Codebook from '@components/Codebook/Codebook';
import Screen from '@components/Screen/Screen';
import { Layout } from '@components/EditorLayout';
import ControlBar from '../ControlBar';
import CollapsableHeader from '../Screen/CollapsableHeader';

/**
 * This component acts as an index for types. i.e. Nodes and Edges,
 * and links to the EditType.
 */
const CodebookScreen = (props) => {
  const {
    layoutId,
    onComplete,
  } = props;

  const buttons = [
    <Button
      key="done"
      onClick={onComplete}
      color="platinum"
    >
      Close
    </Button>,
  ];

  return (
    <Screen
      layoutId={layoutId}
      footer={<ControlBar buttons={buttons} />}
      onComplete={onComplete}
    >
      <CollapsableHeader
        collapsedState={(
          <div className="stage-heading stage-heading--collapsed stage-heading--shadow">
            <Layout>
              <h2>Codebook</h2>
            </Layout>
          </div>
        )}
      >
        <div className="stage-heading stage-heading--inline">
          <Layout>
            <h1 className="screen-heading">Codebook</h1>
            <p>
              Below you can find an overview of the node and edge types that you have
              defined while creating your interview. Entities that are unused may be deleted.
            </p>
          </Layout>
        </div>
      </CollapsableHeader>
      <Layout>
        <Codebook />
      </Layout>
    </Screen>
  );
};

CodebookScreen.propTypes = {
  layoutId: PropTypes.string,
  onComplete: PropTypes.func,
};

CodebookScreen.defaultProps = {
  layoutId: null,
  onComplete: () => {},
};

export default CodebookScreen;
