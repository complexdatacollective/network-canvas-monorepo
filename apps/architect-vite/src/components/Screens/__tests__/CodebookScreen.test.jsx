/* eslint-env jest */

import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import CodebookScreen from '../CodebookScreen';

const mockState = {
  protocol: {
    present: {
      stages: [],
    },
  },
};

const mockProps = {
  store: createStore(() => mockState),
};

describe('<CodebookScreen />', () => {
  it('can render', () => {
    const component = shallow(<CodebookScreen {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
