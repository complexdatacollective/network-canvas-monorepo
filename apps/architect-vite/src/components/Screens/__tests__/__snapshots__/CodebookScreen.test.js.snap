// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodebookScreen /> can render 1`] = `
<Screen
  beforeCloseHandler={null}
  className=""
  footer={
    <ControlBar
      buttons={
        Array [
          <Button
            color="platinum"
            content=""
            disabled={false}
            icon=""
            iconPosition="left"
            onClick={[Function]}
            size=""
            type="button"
          >
            Close
          </Button>,
        ]
      }
      className=""
      secondaryButtons={null}
    />
  }
  header={null}
  layoutId={null}
  onComplete={[Function]}
>
  <CollapsableHeader
    collapsedState={
      <div
        className="stage-heading stage-heading--collapsed stage-heading--shadow"
      >
        <Layout
          className=""
        >
          <h2>
            Codebook
          </h2>
        </Layout>
      </div>
    }
    threshold={115}
  >
    <div
      className="stage-heading stage-heading--inline"
    >
      <Layout
        className=""
      >
        <h1
          className="screen-heading"
        >
          Codebook
        </h1>
        <p>
          Below you can find an overview of the node and edge types that you have defined while creating your interview. Entities that are unused may be deleted.
        </p>
      </Layout>
    </div>
  </CollapsableHeader>
  <Layout
    className=""
  >
    <Connect(Codebook) />
  </Layout>
</Screen>
`;
